### ISSUE TITLE
Cannot read properties of undefined (reading 'filename')

### ISSUE THREAD

**Issue by pbrisbin** [nvuillam/npm-groovy-lint#419]
I'm getting the error in the title attempting to run `npm-groovy-lint --version`.

You can reproduce using a node20 docker image with the following:

```console
cat >package.json <<'EOM'
{
  "name": "repro",
  "version": "0.0.0",
  "dependencies": {
    "npm-groovy-lint": "15.0.0"
  }
}
EOM

cat >Dockerfile <<'EOM'
FROM node:20-slim
RUN mkdir -p /app
WORKDIR /app
COPY package.json .
RUN npm install
EOM

docker build --tag repro .
docker run --rm repro ./node_modules/.bin/npm-groovy-lint --version
```

I get:

```
Unexpected error: Cannot read properties of undefined (reading 'filename')
TypeError: Cannot read properties of undefined (reading 'filename')
    at JavaCaller.manageJavaInstall (/app/node_modules/java-caller/lib/java-caller.js:278:58)
    at async JavaCaller.run (/app/node_modules/java-caller/lib/java-caller.js:95:13)
    at async CodeNarcCaller.startCodeNarcServer (file:///app/node_modules/npm-groovy-lint/lib/codenarc-caller.js:283:28)
    at async CodeNarcCaller.callCodeNarcServer (file:///app/node_modules/npm-groovy-lint/lib/codenarc-caller.js:94:22)
    at async NpmGroovyLint.callCodeNarc (file:///app/node_modules/npm-groovy-lint/lib/groovy-lint.js:275:32)
    at async NpmGroovyLint.run (file:///app/node_modules/npm-groovy-lint/lib/groovy-lint.js:67:13)
    at async NpmGroovyLint.preProcess (file:///app/node_modules/npm-groovy-lint/lib/groovy-lint.js:223:43)
    at async NpmGroovyLint.run (file:///app/node_modules/npm-groovy-lint/lib/groovy-lint.js:65:27)
    at async file:///app/node_modules/npm-groovy-lint/lib/index.js:12:9
```

Omitting `--version` produces the same error. But `--help` seems to work.

Everything was working fine with v14.6.

---

For context, I maintain [Restyled](https://docs.restyled.io/), which wraps this project [as a Restyler](https://github.com/restyled-io/restylers/tree/main/npm-groovy-lint). [Our CI](https://github.com/restyled-io/restylers/actions/runs/10567597025/job/29276899683?pr=805) is failing on the update to v15.

**Comment by nvuillam** [comment]
@pbrisbin thanks for reporting, i'll have a look tonight :)

**Comment by nvuillam** [comment]
I succeeded to reproduce the issue on linux

I'll do what i can to fix, as it's probably in node-java-caller package ^^

**Comment by pbrisbin** [comment]
Thanks!

**Comment by nvuillam** [comment]
@pbrisbin working well now ? :)

**Comment by pbrisbin** [comment]
Yup, v15.0.2 [passed CI](https://github.com/restyled-io/restylers/actions/runs/10624013514/job/29451554886) and we updated our Restyler to it.

**Comment by nvuillam** [comment]
@pbrisbin thanks for your feedback :) 

it seems there is still an issue with latest version, maybe make some tests before using it ^^

https://github.com/nvuillam/npm-groovy-lint/issues/422

**Comment by pbrisbin** [comment]
Ah that's too bad. We do have [some tests](https://github.com/restyled-io/restylers/blob/9a5ff119393415b047e432c1f2b562b9ac8fa96b/npm-groovy-lint/info.yaml#L15-L57), that passed. So the issue must be limited in scope.
